web3/auto/__init__.py

```diff
@@ -0,0 +1,7 @@
+from web3.auto import ipc
+from web3.auto import http
+
+w3 = ipc.ipc()
+
+if not w3:
+    w3 = http.http()
```

Comment:
Right now, this is opening more connections than it needs to:

1. It connects to http even if ipc is available (on module import of http)
2. It re-attempts the connection that was already tested during module import

You could get around both of those issues by switching to a dynamic import and using the w3 already defined, looking something like:

```
for connector in ('ipc', 'http'):
    connection = importlib.import_module('web3.auto.%s' % connector)
    if connection.w3:
        ...
```

---

It could be nice to give these modules a consistent interface, something like `ipc.connect()` and `http.connect()`. Not critical, though.

Reply:
I like the `ipc.connect()` and `http.connect()`, makes it clearer.